src/main/java/com/cloudbees/hudson/plugins/folder/properties/FolderCredentialsProvider.java

```diff
@@ -122,7 +122,20 @@
                                                           @NonNull List<DomainRequirement> domainRequirements) {
         List<C> result = new ArrayList<>();
         Set<String> ids = new HashSet<>();
+
+        // Check if Authentication is authorised to list the credentials
+        Boolean isAuthorised = false;
         if (ACL.SYSTEM.equals(authentication)) {
+            isAuthorised = true;
+        }
+        if (itemGroup instanceof AbstractFolder) { 
+            final AbstractFolder<?> folder = AbstractFolder.class.cast(itemGroup);
+            if (folder.hasPermission(authentication, CredentialsProvider.USE_ITEM)) {
+                isAuthorised = true;
+            }
+        }
+        // Get credentials
+        if (Boolean.TRUE.equals(isAuthorised)) {
```

Comment:
```suggestion
        if (isAuthorised) {
```

Reply:
which I would agree with if not for https://github.com/jenkinsci/cloudbees-folder-plugin/pull/222#discussion_r831064549